src/constructs/static-site/static-site.ts

```diff
@@ -0,0 +1,322 @@
+import { Certificate } from "@aws-cdk/aws-certificatemanager";
+import {
+  CloudFrontWebDistribution,
+  OriginAccessIdentity,
+  ViewerCertificate,
+  ViewerProtocolPolicy,
+} from "@aws-cdk/aws-cloudfront";
+import type { Behavior, CloudFrontWebDistributionProps } from "@aws-cdk/aws-cloudfront/lib/web-distribution";
+import { CanonicalUserPrincipal, PolicyStatement } from "@aws-cdk/aws-iam";
+import { Bucket } from "@aws-cdk/aws-s3";
+import { Annotations, Arn, ArnFormat, CfnOutput, Duration, Token } from "@aws-cdk/core";
+import { Stage } from "../../constants";
+import type { GuDomainNameProps } from "../../types/domain-names";
+import { StageAwareValue } from "../../types/stage";
+import { GuAppAwareConstruct } from "../../utils/mixin/app-aware-construct";
+import { GuCertificate } from "../acm";
+import type { GuStack } from "../core";
+import { AppIdentity } from "../core/identity";
+import { GuCname } from "../dns";
+import { GuS3Bucket } from "../s3";
+
+const GLOBAL_AWS_REGION = "us-east-1";
+const CLOUDFRONT_CERTIFICATE_REQUIREMENT_MESSAGE = `CloudFront requires ACM certificates from the ${GLOBAL_AWS_REGION} region.`;
+const ACM_ARN_RESOURCE = "certificate";
+
+export interface GuStaticSiteProps
+  extends Omit<CloudFrontWebDistributionProps, "originConfigs" | "comment" | "viewerCertificate">,
+    AppIdentity {
+  /**
+   * Domain the CloudFront distribution should be aliased as.
+   *
+   * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-cloudfront-distribution-distributionconfig.html#cfn-cloudfront-distribution-distributionconfig-aliases
+   */
+  domainNameProps: GuDomainNameProps;
+
+  /**
+   * ARN of a pre-existing ACM certificate to attach to CloudFront. It must:
+   *   - exist in the us-east-1 region
+   *   - not have a hard coded account number
+   *
+   * If not provided, a new certificate is created. Note, this means the stack must be deployed in the us-east-1 region.
+   *
+   * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-cloudfront-distribution-viewercertificate.html#cfn-cloudfront-distribution-viewercertificate-acmcertificatearn
+   */
+  preExistingCertificateArn?: StageAwareValue<string>;
```

Comment:
Is there any particular reason that we want to support the usage of pre-existing certificates? 

IIUC it's possible to have more than one certificate with the same domain name and creating this as part of the pattern ensures that it is under version control with the rest of the infrastructure.

I guess by removing this option we'd be forcing people to deploy all static sites into `us-east-1`, but I'm not convinced that's much of a problem.

Reply:
This is a bit of a rabbit hole... In order to fully support the deployment of a static site into `us-east-1` directly, we'd need to [enable `Guardian::DNS::RecordSet` private type into that region](https://github.com/guardian/cfn-private-resource-types/issues/25). https://github.com/guardian/cfn-private-resource-types/pull/24 attempted to do that, but [Riff-Raff wasn't happy](https://riffraff.gutools.co.uk/deployment/view/0da119d3-0c18-4945-b9bf-fbb2370a150b), so we'd need to fix that first.

The `preExistingCertificateArn` option is the ~laziest~ simplest and also means a static site can be defined in the same stack as a Play app (which we run in `eu-west-1`).

Reply:
> ...means a static site can be defined in the same stack as a Play app (which we run in eu-west-1).

I hadn't thought of this, thanks for explaining 👍